packages/jsii/lib/project-info.ts

```diff
@@ -44,6 +44,7 @@ export interface ProjectInfo {
   readonly targets: spec.AssemblyTargets;
   readonly metadata?: { [key: string]: any };
   readonly jsiiVersionFormat: 'short' | 'full';
+  readonly diagnostics?: { readonly [code: string]: string };
```

Comment:
The validation of the values for this map could be done in here (same as we validate the other things - it'd actually be more consistent).

Then you could have:
```suggestion
  readonly diagnostics?: { readonly [code: string]: ts.DiagnosticCategory };
```